docs/guide/en/mapping.md

```diff
@@ -44,6 +44,40 @@ $post = $hydrator->create(Post::class, new ArrayData($data, $map));
 
 This way we take `header` key for `title` and `text` key for `body`.
 
+For nested objects mapping you can use `ObjectMap` class:
+
+```php
+use Yiisoft\Hydrator\ArrayData;
+use Yiisoft\Hydrator\Hydrator;
+use Yiisoft\Hydrator\ObjectMap;
+
+final class Message {
+    public string $subject = '';
+    public ?Body $body = null;
+}
+
+final class Body {
+    public string $text = '';
+    public string $html = '';
+}
+
+$hydrator = new Hydrator();
+
+$data = [
+    'title' => 'Hello, World!',
+    'textBody' => 'Nice to meet you.',
+    'htmlBody' => '<h1>Nice to meet you.</h1>',
+];
+$map = [
+    'subject' => 'title',
+    'body' => new ObjectMap([
+        'text' => 'textBody',
+        'html' => 'htmlBody',    
+    ]), 
```

Comment:
How about constructor, method calls, immutable objects etc.?

Reply:
It has nothing to do with mapping.

We pass key/value array to hydrator. Create/hydrate logic in `Hydrator`, not in mapping.